src/ActionList2/Item.tsx

```diff
@@ -0,0 +1,214 @@
+import React from 'react'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '@radix-ui/react-polymorphic'
+import {useSSRSafeId} from '@react-aria/ssr'
+import {useColorSchemeVar, useTheme} from '../ThemeProvider'
+import Box, {BoxProps} from '../Box'
+import {SxProp, merge} from '../sx'
+import createSlots from '../utils/create-slots'
+import {AriaRole} from '../utils/types'
+import {ListContext} from './List'
+import {customItemThemes} from './hacks'
+import {Selection} from './Selection'
+
+export const getVariantStyles = (variant: ItemProps['variant'], disabled: ItemProps['disabled']) => {
+  if (disabled) {
+    return {
+      color: 'fg.muted',
+      iconColor: 'fg.muted',
+      annotationColor: 'fg.muted'
+    }
+  }
+
+  switch (variant) {
+    case 'danger':
+      return {
+        color: 'danger.fg',
+        iconColor: 'danger.fg',
+        annotationColor: 'fg.muted'
```

Comment:
Are we sure we don't want the annotationColor to be `danger.fg`?

I like how `fg.muted` matches the other annotations
![Screen Shot 2021-11-02 at 3 10 54 PM](https://user-images.githubusercontent.com/2313998/139929602-d1b890b7-69e8-4a6a-a198-d2aeaafae2a9.png)

However, I could also see some value in showing a "danger" color to reinforce that it's a dangerous action:
![Screen Shot 2021-11-02 at 3 10 41 PM](https://user-images.githubusercontent.com/2313998/139933266-7b1b22fe-51fd-45a3-82ff-74f01d23cc15.png)


I'd love to hear if other reviewers have any thoughts on this.